telemetry/telemetry_test.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"os"
+	"testing"
+	"context"
+	"time"
+	"sync"
+	"syscall"
+)
+
+func TestSignalHandler(t *testing.T) {
+	timeoutInterval := 1
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(timeoutInterval) * time.Second)
+	defer cancel()
+
+	reload := make(chan int, 1)
+	testSigChan := make(chan os.Signal, 1)
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+
+	go signalHandler(reload, wg, testSigChan)
+
+	testSigChan <- syscall.SIGTERM
+
+	select {
+	case val := <-reload:
+		if val != 0 {
```

Comment:
Is it possible to get non-zero value here?

Reply:
As of right now, no, but with cert rotation in the future we will use a non-zero value to indicate that we should not exit goroutine but stop server, rotate certs, and restart server.